### PR TITLE
IBX-3944: Reorganized personalization section

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-3944
| Versions      | 4.3, 4.2 master

Scope of this PR:
- removed subcategories: developer guide and best practices
- added landing page
- REST API endpoints moved to API reference subcategory
- language adopted to the style guide

#### Checklist

- [X] Text renders correctly
- [X] Text has been checked with vale
- [X] Description metadata is up to date
- [X] Redirects cover removed/moved pages
- [X] Code samples are working
- [X] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

**Review by nkostiv — APPROVED** [review]

File: docs/personalization/importing_historical_user_tracking_data.md [hunk 1]
```diff
@@ -21,18 +21,18 @@ to the tracking endpoint.
...
-Therefore a signature param needs to be added, which is calculated like following:
+Therefore a signature parameter needs to be added, which is calculated like the following:
 
 `String signature=md5(encode("<itemtype>&<itemid>&<licensekey>&fullprice=<fullprice>&overridetimestamp=<timestamp>&quantity=<quantity>"));`
 
-!!! caution "Params without a value"
+!!! caution "Parameters without a value"
 
-    The order of the params to be signed is crucial.
+    The order of the parameterss to be signed is crucial.
```
Comment (kaborkowski) [inline docs/personalization/importing_historical_user_tracking_data.md:30]: typo?

**Review by kaborkowski — APPROVED** [review]

**Review by dkorostynski — APPROVED** [review]

File: plugins.yml [hunk 2]
```diff
@@ -262,15 +262,15 @@ plugins:
...
-            'guide/personalization/developer_guide/recommendation_api.md': 'personalization/developer_guide/recommendation_api.md'
-            'guide/personalization/developer_guide/content_api.md': 'personalization/developer_guide/content_api.md'
-            'guide/personalization/developer_guide/user_api.md': 'personalization/developer_guide/user_api.md'
-            'guide/personalization/developer_guide/attributes_search_elastic.md': 'personalization/developer_guide/attribute_search_in_elasticsearch.md'
-            'guide/personalization/best_practices/tracking_integration.md': 'personalization/best_practices/tracking_integration.md'
-            'guide/personalization/best_practices/recommendation_integration.md': 'personalization/best_practices/recommendation_integration.md'
+            'guide/personalization/developer_guide/tracking_api.md': 'personalization/api_reference/tracking_api.md'
+            'guide/personalization/developer_guide/importing_historical_user_tracking_data.md': 'personalization/importing_historical_user_tracking_data.md'
```
Comment (dabrt) [inline plugins.yml:266]:
```suggestion
            'guide/personalization/developer_guide/importing_historical_user_tracking_data.md': 'personalization/event_tracking/import_historical_user_tracking_data.md'
```

File: plugins.yml [hunk 3]
```diff
@@ -262,15 +262,15 @@ plugins:
...
+            'guide/personalization/developer_guide/tracking_with_yct.md': 'personalization/tracking_with_yct.md'
```
Comment (dabrt) [inline plugins.yml:267]:
```suggestion
            'guide/personalization/developer_guide/tracking_with_yct.md': 'personalization/event_tracking/tracking_with_yct.md'
```

File: plugins.yml [hunk 4]
```diff
@@ -262,15 +262,15 @@ plugins:
...
+            'guide/personalization/developer_guide/recommendation_api.md': 'personalization/api_reference/recommendation_api.md'
+            'guide/personalization/developer_guide/content_api.md': 'personalization/api_reference/content_api.md'
+            'guide/personalization/developer_guide/user_api.md': 'personalization/api_reference/user_api.md'
+            'guide/personalization/developer_guide/attributes_search_elastic.md': 'personalization/attribute_search_in_elasticsearch.md'
+            'guide/personalization/best_practices/tracking_integration.md': 'personalization/tracking_integration.md'
```
Comment (dabrt) [inline plugins.yml:272]:
```suggestion
            'guide/personalization/best_practices/tracking_integration.md': 'personalization/event_tracking/tracking_methods.md'
```

File: plugins.yml [hunk 5]
```diff
@@ -262,15 +262,15 @@ plugins:
...
+            'guide/personalization/best_practices/recommendation_integration.md': 'personalization/recommendation_integration.md'
```
Comment (dabrt) [inline plugins.yml:273]:
```suggestion
            'guide/personalization/best_practices/recommendation_integration.md': 'personalization/integrate_recommendation_results.md'
```

File: docs/personalization/importing_historical_user_tracking_data.md [hunk 6]
```diff
@@ -21,18 +21,18 @@ to the tracking endpoint.
...
-Therefore a signature param needs to be added, which is calculated like following:
+Therefore a signature parameter needs to be added, which is calculated like the following:
 
 `String signature=md5(encode("<itemtype>&<itemid>&<licensekey>&fullprice=<fullprice>&overridetimestamp=<timestamp>&quantity=<quantity>"));`
 
-!!! caution "Params without a value"
+!!! caution "Parameters without a value"
 
-    The order of the params to be signed is crucial.
+    The order of the  to be signed is crucial.
```
Comment (julitafalcondusza) [inline docs/personalization/importing_historical_user_tracking_data.md:30]: The order of the parameters to be signed is crucial.

**Review by julitafalcondusza — COMMENTED** [review]
30. "parameters" missing

File: docs/personalization/api_reference/api_reference.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
+description: Correlate metadata with user data and combine users into clusters of certain type.
+---
+
+# API reference
+
+This secton provides documentation for the Personalization API operations.
```
Comment (julitafalcondusza) [inline docs/personalization/api_reference/api_reference.md:7]: This section provides documentation for the Personalization API operations.

**Review by julitafalcondusza — COMMENTED** [review]
7. This section (...).

**Review by julitafalcondusza — APPROVED** [review]

**Review by dabrt — APPROVED** [review]

**Pull request merged**